### ISSUE TITLE
RemoveBuildingForPlayer

### ISSUE THREAD

**Issue by qaisjp** [qaisjp/mta-amx#2]
RemoveBuildingForPlayer is currently stubbed at https://github.com/multitheftauto/amx/blob/e083c26cfbc89f69e6bbc84e6db7d9c3c6a1df25/amx/server/syscalls.lua#L774-L777

MTA:

- [removeWorldModel](https://wiki.multitheftauto.com/wiki/RemoveWorldModel)
- ```lua
  bool removeWorldModel ( int modelID, float radius, float x, float y, float z [, int interior = 0 ] )
  ```

SAMP:

- [RemoveBuildingForPlayer](https://wiki.sa-mp.com/wiki/RemoveBuildingForPlayer)
- ```pawn
  RemoveBuildingForPlayer(playerid, modelid, Float:fX, Float:fY, Float:fZ, Float:fRadius)
  ```

Differences:

- SAMP's native does not return anything.
- SAMP's native does not have an interior parameter. I am not sure if it affects interior 0 or all interiors. 
- SAMP's native allows you to give `-1` for the model ID to remove all objects within the specified radius. See example on MTA wiki to see how this can be implemented in Lua.

**Comment by theSpool** [comment]
Nice, I was wondering if there was a function in MTA to do this, obviously there was but I couldn't just find it, I was about to ask but saw this. Thanks for all the details.

> SAMP's native does not have an interior parameter. I am not sure if it affects interior 0 or all interiors.

It affects all interiors. I'd guess we'd then have to do a loop just like in the model ID case, right?

SA-MP also has this bug where the Buildings are not restored on reconnection, so if the mode changes, the previous gamemodes removed buildings will stay. I assume this is not the case in MTA. It's not like any modes depend on this buggy behaviour though.

**Comment by qaisjp** [comment]
yep!

**Comment by qaisjp** [comment]
> yep!

nope.

> It affects all interiors. I'd guess we'd then have to do a loop just like in the model ID case, right?

The MTA documentation was wrong. Initially it was documented as:

```lua
bool removeWorldModel ( int modelID, float radius, float x, float y, float z [, int interior = 0 ] )
```

But the default value for `interior` is actually `-1`!

So f4a959e98a6dca72ca17780280ed0e49bbb51968 will work fine to delete stuff in all interiors.

That commit still needs to account for modelID `-1`, perhaps later I could just add `-1` support to MTA, but for now a for-loop is needed.

/cc @colistro123

**Comment by colistro123** [comment]
This has been added, closing this now I guess

**Comment by qaisjp** [comment]
added here https://github.com/multitheftauto/amx/commit/29cf87ba3af0a44109bc5549dd4055e45e2d5b13#diff-618b9e8da428b783246c66867232bfa3R402-R411 and typo fixed here 6ab1c9a9fb12975c9abf4cb2be97889fae11a205
